create.ps1

```diff
@@ -171,74 +154,93 @@ try {
         }
 
-        Write-Information 'Lookup Department via Mapping'
         try {
-            $mappedDepartment = Get-MappedValueFromMappingFile -CsvFileLocation $actionContext.Configuration.DepartmentMappingFile -ContractPropertyExternalId $actionContext.Data.mapping.departmentId -CsvPropertyHeaderName "Department"
-            $department = Get-FacilitorResource -Url "$($actionContext.Configuration.BaseUrl)/api2/departments?id=$($mappedDepartment.FacilitorDepartmentId)" -Property "department" -PropertyId $mappedDepartment.FacilitorDepartmentId
+            $splatGetDepartments = @{
+                Url            = "$($actionContext.Configuration.BaseUrl)/api2/departments?name=$($actionContext.Data.mapping.departmentId)"
+                Property       = 'departments'
+                PropertyId     = $actionContext.Data.mapping.departmentId
+                SearchProperty = 'name'
+            }
+            $department = Get-FacilitorResource @splatGetDepartments
 
             $departmentValue = @{
                 id = $department.id
             }
-            $actionContext.Data | Add-Member -MemberType NoteProperty -Name 'department' -Value $departmentValue
-        } catch {
+            $actionContext.Data | Add-Member -MemberType NoteProperty -Name "department" -Value $departmentValue
+        }
+        catch {
             $isValidationError = $true
-            Add-AuditLog -ErrorMessage "$($_.Exception.message)" -IsError $true
+            Add-AuditLog -ErrorMessage "$($_.Exception.message)" -IsError $true             
         }
 
-        Write-Information 'Lookup Location via Mapping'
         try {
-            $mappedLocation = Get-MappedValueFromMappingFile -CsvFileLocation $actionContext.Configuration.LocationMappingFile -ContractPropertyExternalId $actionContext.Data.mapping.locationId -CsvPropertyHeaderName "Location"
-            $location = Get-FacilitorResource -Url "$($actionContext.Configuration.BaseUrl)/api2/locations?id=$($mappedLocation.FacilitorlocationId)" -Property "location" -PropertyId $mappedLocation.FacilitorLocationId
+            $splatGetLocations = @{
+                Url            = "$($actionContext.Configuration.BaseUrl)/api2/locations?visitzipcode=$($actionContext.Data.mapping.locationId)"
+                Property       = 'locations'
+                PropertyId     = $actionContext.Data.mapping.locationId
+                SearchProperty = 'visitzipcode'
+            }
+            $location = Get-FacilitorResource @splatGetLocations
+
             $actionContext.Data.custom_fields = @(
                 [PSCustomObject]@{
-                    propertyid = 1080
+                    propertyid = 1060
                     value      = "$($location.id)"
-                    Type       = 'N'
+                    Type       = "N"
                     sequence   = 50
-                    label      = 'Locatie ID'
+                    label      = "Locatie ID"
```

Comment:
Meer ter info je hoeft dit niet aan te passen. Als je dubbele quotes gebruikt gaat powershell alles proberen te resolven dus als er variabele of  dergelijke inzitten dan pakt hij daar de waarde van. In dit geval heb je er alleen tekst in zitten en is dat dus niet nodig. scheelt wat resources

Reply:
Geen idee waarom dit is aangepast. Normaal let ik daar wel op.